HIP/hip-904-frictionless-airdrop.md

```diff
@@ -0,0 +1,714 @@
+
+```protobuf
+/**
+* A token airdrop transactions change on state is dependent on teh recieving accounts assocaition with a token or the presence of avialable automatics associations.
+* Air drops that resuly in transfers will see the transfers populated in the record token_transfer_list.
+* Airdrops that consume an assocaition slot will eb represented in the record automatioc_association field.
+* Airdrops that result in pending trasnfers will see the intended trasnfer captures in the record pending_airdrops field.
+*/ 
+message TokenAirdropTransactionBody {
+    /**
+    * The desired list of tokens to airdrop. If any custom fees are assessed,
+    * the ledger will try to deduct them from the payer of this airdrop,
+    * resolving the transaction to INSUFFICIENT_PAYER_BALANCE_FOR_CUSTOM_FEE if
+    * this is not possible.
+    * The sender must have all the succifient balance at the time of the transaction to support this transaction.
+    *
+    * If the receiver account has already associated with the given tokens, then
+    * the fee to transfer will be identical to a token transfer. If the receiver
+    * has open auto-association slots but has not yet associated with the token,
+    * then the payer will pay for the transfer, association, one renewal credit,
+    * and airdrop fee. If the receiver has not associated with the token and has
+    * no open association slot, then the transfer will not deliver the token,
+    * but will create a `pending airdrop` instead, with the receiver account
+    * listed on the `pending airdrop`. The payer will pay for the transfer,
+    * association, one renewal credit, and airdrop fee.
+    * All tokenTransferList items must succeed (transfer of pedning state) for the transaction to succeed.
+    */
+    repeated TokenTransferList tokenTransfers = 1;
+}
+```
+
+TokenAirdrop transaction will support the transfers of value in a similar manner to CryptoTransfer with the difference of only supporting tokens (no HBAR transfer support) and supporting the resulting pending transfer state where a CryptoTransfer would have failed due to the lack of association options.
+
+However, it is important to have a separate HAPI transaction type for a few reasons
+
+- the ability to call out the potential separate behaviours based on maxAutomaticAssociation, whereby transfer can end up in a pending transfer state or a consumption of an association slot.
+- cleaner conceptual understanding
+- naming allows the grouping of functionality with Cancel and Claim which complete the pending state a TokenAirdrop may creates
+- explicit specification in record stream.
+- improved developer experience
+
+Similar restrictions around transfer list and aggregation across sender and receiver from CryptoTransfer will be applied to TokenAirdop e.g. max 10 transfers per transaction
+
+At the time of this HIP `TokenAirdropTransaction` will not be supported by the Scheduled Transaction Service. A future HIP may explore this support.
```

Comment:
Should these limits be specified (also) in a different section, i.e., above somewhere with the rest of the textual specification?  The scheduled Tx limitation seems less of a "HAPI" section item, in particular.

Should the max 10 transfers per transaction limit be specified in the message's javadoc?

(Are protobuf specifications in HIPs meant to be applied directly to the protobuf source files, and if so does that include comments?  I don't know.  I do know there's an effort going on right now to make the protobuf sources' javadoc be more rigorously a specification and documentation source.)

Reply:
Added the max 10 limit in the proto comments

Reply:
Still thinking on the separate section